File: CS/Q253493/FilterControl/CustomFilterControl.cs
using System;
using DevExpress.Data;
using DevExpress.XtraEditors.Filtering;
using DevExpress.Data.Filtering.Helpers;

namespace Q253493.FilterControl {
    public class CustomFilterControl :DevExpress.XtraEditors.FilterControl {
        protected override WinFilterTreeNodeModel CreateModel() {
            return new CustomWinFilterTreeNodeModel(this);
        }

        void RaiseGetDefaultOperation(GetDefaultOperationEventArgs args) {
            EventHandler<GetDefaultOperationEventArgs> handler = Events[fGetDefaultOperation]
                as EventHandler<GetDefaultOperationEventArgs>;
            if (handler != null)
                handler(this, args);
        }

        internal ClauseType GetDefaultOperationCore(IBoundProperty property, ClauseType operation) {
            GetDefaultOperationEventArgs args = new GetDefaultOperationEventArgs(property, operation);
            RaiseGetDefaultOperation(args);
            return args.ClauseType;
        }

        static readonly object fGetDefaultOperation = new object();
        public event EventHandler<GetDefaultOperationEventArgs> GetDefaultOperation {
            add { Events.AddHandler(fGetDefaultOperation, value); }
            remove { Events.RemoveHandler(fGetDefaultOperation, value); }
        }
    }
}

File: VB/Q253493/FilterControl/CustomFilterControl.vb
Imports System
Imports DevExpress.Data
Imports DevExpress.XtraEditors.Filtering
Imports DevExpress.Data.Filtering.Helpers

Namespace Q253493.FilterControl
    Public Class CustomFilterControl
        Inherits DevExpress.XtraEditors.FilterControl

        Protected Overrides Function CreateModel() As WinFilterTreeNodeModel
            Return New CustomWinFilterTreeNodeModel(Me)
        End Function

        Private Sub RaiseGetDefaultOperation(ByVal args As GetDefaultOperationEventArgs)
            Dim handler As EventHandler(Of GetDefaultOperationEventArgs) = TryCast(Events(fGetDefaultOperation), EventHandler(Of GetDefaultOperationEventArgs))
            If handler IsNot Nothing Then
                handler(Me, args)
            End If
        End Sub

        Friend Function GetDefaultOperationCore(ByVal [property] As IBoundProperty, ByVal operation As ClauseType) As ClauseType
            Dim args As New GetDefaultOperationEventArgs([property], operation)
            RaiseGetDefaultOperation(args)
            Return args.ClauseType
        End Function

        Private Shared ReadOnly fGetDefaultOperation As New Object()
        Public Custom Event GetDefaultOperation As EventHandler(Of GetDefaultOperationEventArgs)
            AddHandler(ByVal value As EventHandler(Of GetDefaultOperationEventArgs))
                Events.AddHandler(fGetDefaultOperation, value)
            End AddHandler
            RemoveHandler(ByVal value As EventHandler(Of GetDefaultOperationEventArgs))
                Events.RemoveHandler(fGetDefaultOperation, value)
            End RemoveHandler
            RaiseEvent(ByVal sender As System.Object, ByVal e As GetDefaultOperationEventArgs)
            End RaiseEvent
        End Event
    End Class
End Namespace
